libs/config_registry/include/hpx/config_registry.hpp

```diff
@@ -10,17 +10,33 @@
 #include <string>
 #include <vector>
 
+#if (defined(_WIN32) || defined(__WIN32__) || defined(WIN32))
+#if defined(HPX_CONFIG_REGISTRY_EXPORTS)
+#define HPX_CONFIG_REGISTRY_EXPORT __declspec(dllexport)
+#else
+#define HPX_CONFIG_REGISTRY_EXPORT __declspec(dllimport)
+#endif
+#elif defined(__NVCC__) || defined(__CUDACC__)
+#define HPX_CONFIG_REGISTRY_EXPORT /* empty */
+#else
+#if defined(HPX_CONFIG_REGISTRY_EXPORTS)
+#define HPX_CONFIG_REGISTRY_EXPORT __attribute__((visibility("default")))
+#else
+#define HPX_CONFIG_REGISTRY_EXPORT /* empty */
+#endif
```

Comment:
Should we have a global fallback in place:
```suggestion
#if defined(HPX_CONFIG_REGISTRY_EXPORTS)
#define HPX_CONFIG_REGISTRY_EXPORT __attribute__((visibility("default")))
#endif
#if !defined(HPX_CONFIG_REGISTRY_EXPORT)
#define HPX_CONFIG_REGISTRY_EXPORT /* empty */
#endif
```